### PR TITLE
API for creating and connecting DCs

### PR DESCRIPTION
**Pull request by deepthidevaki**
This pull request provides API for joining nodes to form a DC and connecting multiple DCs for replication. The common tests in test/* use the new interface to set up the DCs.

Related changes to protocol buffer specifications are in riak_pb branch antidote_cluster_management_api.

### CODE REVIEW

File: docs/pages/mydoc/rawapi.md [hunk 1]
```diff
@@ -125,3 +125,25 @@ It is not possible to read and update in the same transaction.
...
+## Cluster management
+
+An Antidote data center (DC) is a cluster of multiple antidote nodes. Antidote deployment can have multiple DCs where each DCs is a full replica. The cluster management API provides functions to create DCs and connect them.
+
+#### Example ####
+If 6 antidote nodes have already started, we can create two DCs with 3 nodes each as follows:
+
+```erlang
+     rpc:call('antidote@node1', create_dc, [['antidote@node1', 'antidote@node2', 'antidote@node3']]),
+     rpc:call('antidote@node4', create_dc, [['antidote@node4', 'antidote@node5', 'antidote@node6']]).
```
Comment (bieniusa) [inline docs/pages/mydoc/rawapi.md]: 1) What happens if two nodes are added for different DCs?
2) Does every node need to be associated to a DCs?

File: docs/pages/mydoc/rawapi.md [hunk 2]
```diff
@@ -125,3 +125,25 @@ It is not possible to read and update in the same transaction.
...
+```
+
+To connect two DCs:
+
+```erlang
+     {ok, Descriptor1} = rpc:call('antidote@node1', get_connection_descriptor, []),
+     {ok, Descriptor2} = rpc:call('antidote@node4', get_connection_descriptor, []),
+     Descriptors = [Descriptor1, Descriptor2],
+     rpc:call('antidote@node1', subscribe_updates_from, [Descriptors]),
```
Comment (bieniusa) [inline docs/pages/mydoc/rawapi.md]: What happens if only one of the nodes subscribes for updates? Might be a typical error.
Would it make sense to have a function for this connecting functionality (taking as parameter the DC some nodes)?
Actually, can we use arbitrary nodes for connecting the corresponding DCs?

**Comment by deepthidevaki** [comment]
@albsch Changes in this branch was conflicting with your changes to test_utils. I resolved them. Can you check if I removed any optimizations that you did?

**Comment by albsch** [comment]
Everything is ok, branch can get merged.

**Pull request merged**